org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/xtext/ui/JdtHyperlinkFactory.java

```diff
@@ -34,7 +35,18 @@ public boolean canHandle(EObject to) {
 		return to instanceof JvmIdentifiableElement && to.eResource() instanceof TypeResource;
 	}
 
+	/**
+	 * @deprecated use {@link #createHyperlink(IRegion, EObject, IHyperlinkAcceptor)} instead.
+	 */
+	@Deprecated
```

Comment:
I share the concerns raised by @cdietrich - this change is hard to test for backwards compatibility since createHyperlink may both be called or overridden by clients.
In fact I would prefer the new, correct signature to delegate to the old, now deprecated signature if the type matches. Otherwise do it's own thingy.